pkgs/applications/window-managers/miriway/default.nix

```
@@ -0,0 +1,78 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, unstableGitUpdater
+, nixosTests
+, cmake
+, pkg-config
+, mir
+, libxkbcommon
+}:
+
+stdenv.mkDerivation rec {
+  pname = "miriway";
+  version = "unstable-2022-12-18";
+
+  src = fetchFromGitHub {
+    owner = "Miriway";
+    repo = "Miriway";
+    rev = "d294c303cb99b7becb0d6686be9a09f0a1f57596";
+    hash = "sha256-H+IZgI1IQxNl5yAygbDKXkyXajGHV/mp9gEqZcp0TeE=";
+  };
+
+  postPatch = ''
+    substituteInPlace CMakeLists.txt \
+      --replace "\''${CMAKE_INSTALL_PREFIX}/bin" "\''${CMAKE_INSTALL_BINDIR}" \
+      --replace "/usr/share" "\''${CMAKE_INSTALL_DATADIR}" \
+      --replace "/etc" "\''${CMAKE_INSTALL_SYSCONFDIR}"
+
+    sed -i -e '/project(/a include(GNUInstallDirs)' CMakeLists.txt
+  '';
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ];
+
+  buildInputs = [
+    mir
+    libxkbcommon
+  ];
+
+  passthru = {
+    updateScript = unstableGitUpdater { };
+    providedSessions = [ "miriway" ];
+    tests.vm = nixosTests.miriway;
```

This seems to be the convention.
```suggestion
    inherit (nixosTests) miriway;
```

---

Agree, will change.

---

I forgot the surrounding `tests = {}` in the initial comment.

---

Whoops, should be correct now.